components/infobox/wikis/starcraft2/infobox_person_custom_shared.lua

```diff
@@ -0,0 +1,176 @@
+	['content producer'] = 'Content producer', ['streamer'] = 'false',
+}
+
+local _raceData
+local _statusStore
+local _militaryStore
+local _args
+
+local _PAGENAME = mw.title.getCurrentTitle().prefixedText
+
+local CustomPerson = {}
+
+function CustomPerson.setArgs(args)
+	_args = args
+end
+
+function CustomPerson.shouldStoreData()
+	if
+		_args.disable_smw == 'true' or _args.disable_lpdb == 'true' or _args.disable_storage == 'true'
+		or Variables.varDefault('disable_SMW_storage', 'false') == 'true'
+		or mw.title.getCurrentTitle().nsText ~= ''
+	then
+		Variables.varDefine('disable_SMW_storage', 'true')
+		return false
+	end
+	return true
+end
+
+function CustomPerson.nameDisplay()
+	CustomPerson.getRaceData(_args.race or 'unknown')
+	local raceIcon = RaceIcon({'alt_' .. _raceData.race})
+	local name = _args.id or _PAGENAME
+
+	return raceIcon .. '&nbsp;' .. name
+end
+
+function CustomPerson.getRaceData(race)
+	race = string.lower(race)
+	race = CleanRace[race] or race
+	local display = CustomPerson.raceDisplayLookupTable[race]
+	if not display and race ~= 'unknown' then
+		display = '[[Category:InfoboxRaceError]]<strong class="error">' ..
+			mw.text.nowiki('Error: Invalid Race') .. '</strong>'
+	end
+
+	_raceData = {
+		race = race,
+		faction = _FACTION1[race] or '',
+		faction2 = _FACTION2[race] or '',
+		display = display
+	}
+
+	return display
+end
+
+function CustomPerson.adjustLPDB(_, lpdbData)
+	local extradata = lpdbData.extradata or {}
+	extradata.race = _raceData.race
+	extradata.faction = _raceData.faction
+	extradata.faction2 = _raceData.faction2
+	extradata.lc_id = string.lower(_PAGENAME)
+	extradata.teamname = _args.team
+	extradata.role = _args.role
+	extradata.role2 = _args.role2
+	extradata.militaryservice = _militaryStore
+	extradata.activeplayer = (not _statusStore) and Variables.varDefault('isActive', '') or ''
+
+	if Variables.varDefault('racecount') then
+		extradata.racehistorical = true
+		extradata.factionhistorical = true
+	end
+
+	lpdbData.extradata = extradata
+
+	return lpdbData
+end
+
+function CustomPerson.military(military)
+	if military and military ~= 'false' then
+		local display = military
+		military = string.lower(military)
+		local militaryCategory = ''
+		if String.contains(military, 'starting') or String.contains(military, 'pending') then
+			militaryCategory = '[[Category:Persons waiting for Military Duty]]'
+			_militaryStore = 'pending'
+		elseif
+			String.contains(military, 'ending') or String.contains(military, 'started')
+			or String.contains(military, 'ongoing')
+		then
+			militaryCategory = '[[Category:Persons on Military Duty]]'
+			_militaryStore = 'ongoing'
+		elseif String.contains(military, 'fulfilled') then
+			militaryCategory = '[[Category:Persons expleted Military Duty]]'
+			_militaryStore = 'fulfilled'
+		elseif String.contains(military, 'exempted') then
+			militaryCategory = '[[Category:Persons exempted from Military Duty]]'
+			_militaryStore = 'exempted'
+		end
```

Comment:
Can be converted to a table

Reply:
you mean something like the following?
```lua
		local _MILITARY_DATA = {
			starting = {category = 'Persons waiting for Military Duty', stroeValue = 'pending'},
			pending = {category = 'Persons waiting for Military Duty', stroeValue = 'pending'},
			ending = {category = 'Persons on Military Duty', stroeValue = 'ongoing'},
			started = {category = 'Persons on Military Duty', stroeValue = 'ongoing'},
			fulfilled = {category = 'Persons expleted Military Duty', stroeValue = 'fulfilled'},
			exempted = {category = 'Persons exempted from Military Duty', stroeValue = 'exempted'},
		}
		.....
		for key, item in pairs(_MILITARY_DATA) do
			if String.contains(military, key) then
				militaryCategory = '[[Category:' .. item.category .. ']]'
				_militaryStore = item.stroeValue
				break
			end
		end
```

Reply:
Yeah, though you can have multiple keys to same data instead of repeating
(note misspelled store)

Reply:
> you can have multiple keys to same data instead of repeating

how would i define them in one go?
wouldn't it need to be done in 2 steps?
```lua
local _MILITARY_DATA = {
	starting = {category = 'Persons waiting for Military Duty', storeValue = 'pending'},
	ending = {category = 'Persons on Military Duty', storeValue = 'ongoing'},
	fulfilled = {category = 'Persons expleted Military Duty', storeValue = 'fulfilled'},
	exempted = {category = 'Persons exempted from Military Duty', storeValue = 'exempted'},
}
_MILITARY_DATA.pending = _MILITARY_DATA.starting
_MILITARY_DATA.started = _MILITARY_DATA.ending
```

Reply:
It would be possible for the key side of the table to be a table as well. 
However what you have works just fine imo

Not sure if on purpose or by mistake, you don't have "ongoing" in neither of your tables.

Reply:
thx good catch